src/main/java/com/iota/iri/utils/log/StatusLogger.java

```diff
@@ -0,0 +1,133 @@
+package com.iota.iri.utils.log;
+
+import org.slf4j.Logger;
+
+/**
+ * This class represents a wrapper for the {@link Logger} used by IRI that implements a logic to regularly issue status
+ * updates about a certain task.
+ *
+ * Instead of printing all messages immediately and unnecessarily spamming the console output, it rate limits the amount
+ * of messages shown and only prints messages if the status has changed since the last output.
+ */
+public class StatusLogger {
+    /**
+     * Holds the interval in milliseconds that status messages get printed.
+     */
+    private static final int LOG_INTERVAL = 3000;
+
+    /**
+     * Holds a reference to the underlying logger.
+     */
+    private final Logger logger;
+
+    /**
+     * Holds the current status message that shall get printed to the console.
+     */
+    private String statusMessage = "";
+
+    /**
+     * Holds the last status message that was printed.
+     */
+    private String lastPrintedStatusMessage = "";
+
+    /**
+     * Holds a reference to the thread that takes care of scheduling the next status update.
+     */
+    private Thread outputThread = null;
+
+    /**
+     * Holds a timestamp value when the last message was printed.
+     */
+    private long lastLogTime = 0;
+
+    /**
+     * Constructor of the class.
+     *
+     * It simply stores the passed in parameters to be able to access them later on.
+     *
+     * @param logger {@link Logger} object that is used in combination with logback to issue messages in the console
+     */
+    public StatusLogger(Logger logger) {
+        this.logger = logger;
+    }
+
+    /**
+     * This method allows us to set a new status that shall get printed to the screen.
+     *
+     * It checks if the given message is new and then either prints the message immediately (if enough time has passed
+     * since the last message) or spawns a {@link Thread} that will print the new status after the given timeout.
+     *
+     * When creating the {@link Thread} it copies its name so the output is "transparent" to the user and the effect
+     * is the same as dumping the message manually through the {@link Logger} object itself.
+     *
+     * @param message status message that shall get printed
+     */
+    public void status(String message) {
+        if (!message.equals(statusMessage)) {
+            statusMessage = message;
+
+            if (System.currentTimeMillis() - lastLogTime >= LOG_INTERVAL) {
+                printStatusMessage();
+            } else if (outputThread == null) {
```

Comment:
This could lead to some messages going to /dev/null: if the `outputThread` is already sleeping to deliver a message, all messages in the meanwhile would get lost. If you change for `else` then this method will synchronize calls (assuming status messages are different), making this logger a bottleneck on `LOG_INTERVAL`.

Reply:
Thats the whole point of this class - to not show "all" messages but to just dump a message every 3 seconds with the "latest" status.

Since we store the next status message that shall be printed a few lines before in:

statusMessage = message;

The output thread will print this last message and drop the ones that happened in between.

Reply:
> Thats the whole point of this class - to not show "all" messages but to just dump a message every 3 seconds with the "latest" status.

I thought the purpose of the class if how you described in the JavaDoc:

> It checks if the given message is new and then either prints the message immediately (if enough time has passed since the last message)

Reply:
> Thats the whole point of this class - to not show "all" messages but to just dump a message every 3 seconds with the "latest" status.

I've understood something different from the JavaDoc:

> It checks if the given message is new and then either prints the message immediately (if enough time has passed since the last message)

but then again I don't see the point of this if:
`if (!message.equals(statusMessage)) {`
if you always disregard the previous statusMessage anyway, either equal or not.

Reply:
If the "new" message is the same as the current status message it was either printed already or is scheduled to be printed, so we can ignore it.

Since with this check only, the message can change to sth else and then back to what it was before we have an additional check in the print method to not print the same status update twice.